### ISSUE TITLE
`TaintedExtract` should not be thrown for array values (only keys)

### ISSUE THREAD

**Issue by gharlan** [vimeo/psalm#11263]
```php
extract(['foo' => $_GET['foo']]);
```

This throws a `TaintedExtract` error. In my opinion `TaintedExtract` should only look at the array keys (used as variable names). The given code is the same as:

```php
$foo = $_GET['foo'];
```

which is safe.

While this would be dangerous:

```php
extract([$_GET['foo'] => 'foo']);
```

---

https://psalm.dev/r/6620d37bc4

* `TaintedExtract` for line 3 is correct
* `TaintedExtract` for line 5 is wrong in my opinion. There is nothing wrong in this line. If the variable `$foo` is used in any other taint sink, it is reported anyway (see line 6).
* So `TaintedHtml` and `TaintedTextWithQuotes` for line 6 are also correct

**Comment by orklah** [comment]
I guess if every key for an array is known and the array is sealed (meaning no other keys could be in the array) it should be safe to not emit a TaintedExtract as long as the taint is transmitted to every new variable created by extract
